specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/cosmos-db.json

```diff
@@ -4298,6 +4298,450 @@
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableSqlDatabases/{restorableSqlDatabaseId}": {
+      "get": {
+        "operationId": "RestorableSqlDatabases_Get",
+        "x-ms-examples": {
+          "CosmosDBRestorableSqlDatabaseGet": {
+            "$ref": "./examples/CosmosDBRestorableSqlDatabaseGet.json"
+          }
+        },
+        "description": "Get the properties of a restorable SQL database.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          },
+          {
+            "$ref": "#/parameters/instanceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/restorableSqlDatabaseIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The restorable SQL databases properties were retrieved successfully.",
+            "schema": {
+              "$ref": "#/definitions/RestorableSqlDatabaseGetResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DocumentDB/locations/{location}/restorableDatabaseAccounts/{instanceId}/restorableSqlContainers": {
+      "get": {
+        "operationId": "RestorableSqlContainers_List",
+        "x-ms-examples": {
+          "CosmosDBRestorableSqlContainerList": {
+            "$ref": "./examples/CosmosDBRestorableSqlContainerList.json"
+          }
+        },
+        "description": "Lists all the restorable Azure Cosmos DB SQL containers available for a specific database.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          },
+          {
+            "$ref": "#/parameters/instanceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/restorableSqlDatabaseRidParameter"
```

Comment:
A collection GET with a required query parameter is not going to be consumable by downstream services like ARG and Policy.  Why was this moved from a child resource of restorableSqlDatabases to its own GET with a required query param?  At minimum the query param should be optional and show containers from all databases

Reply:
Thanks.. updated the spec to make these parameters optional